### ISSUE TITLE
price_excl_tax is not required for when you don't track stock

### ISSUE THREAD

**Issue by radeinla** [django-oscar/django-oscar#1795]
Why is it that way? I don't get the reason behind why its not required if you are not tracking stock?

Here is the relevant code:

```python
class StockRecordForm(forms.ModelForm):

    def __init__(self, product_class, user, *args, **kwargs):
        # The user kwarg is not used by stock StockRecordForm. We pass it
        # anyway in case one wishes to customise the partner queryset
        self.user = user
        super(StockRecordForm, self).__init__(*args, **kwargs)

        # If not tracking stock, we hide the fields
        if not product_class.track_stock:
            del self.fields['num_in_stock']
            del self.fields['low_stock_threshold']
        else:
            self.fields['price_excl_tax'].required = True
            self.fields['num_in_stock'].required = True

    class Meta:
        model = StockRecord
        exclude = ('product', 'num_allocated')

```

**Comment by john-parton** [comment]
That does seem confusing. I think it's to allow for a hook to compute the price in some other way, but I really don't know.

**Comment by radeinla** [comment]
Yeah that seems to be the case. We just removed the if-del from our fork. I just wanted to raise this concern. We'd love to do a PR but this seems to be a 5 minute fix for you guys so I created a GH issue instead.

**Comment by balsagoth** [comment]
+1
Because of this when rebuild search index it raises an error.

`TypeError: unsupported operand type(s) for +: 'NoneType' and 'Decimal'`

**Comment by maikhoepfel** [comment]
I agree this looks like a bug. Marking it as such. I'm afraid I don't have time to look into this now, but it should be easy to override this form via the usual mechanisms.

**Comment by sasha0** [comment]
I'll look into this.

**Comment by sasha0** [comment]
Per comment to StockRecord.price_excl_tax field:
> It is NULLable because some items don't have a fixed price but require a runtime calculation (possible from an external service).

So, I suppose this is the case for not tracking case, but @codeinthehole knows the best, since he implemented this code.

> ```TypeError: unsupported operand type(s) for +: 'NoneType' and 'Decimal'```

@balsagoth could you please paste here traceback of the exception? Thanks.

**Comment by sasha0** [comment]
As mentioned above, that could be useful for dynamic price calculation, fetching prices from external service etc. Basket-related issues has been addressed by PR #2283, as for the issue with search index rebuilt - could re-create by myself for tax-inclusive pricing strategy.
